### PR TITLE
Refactor: replace moment.js with day.js

### LINKED ISSUE #74: Refactor: replace moment.js with day.js
The frontend client uses [Moment.js](https://momentjs.com/docs/#/-project-status/) for pretty much anything date related however it now considered a __legacy__ project and advises users to migrate to a new library. 

[Day.js](https://github.com/iamkun/dayjs) seems like the best library with a good API.

All instances of moment.js should be replaced with day.js calls while keeping the same functionality.

__For hacktoberfest users__

This issue touches many different places within the frontend code, so I would only recommend this to users with a good idea on how to use React & Typescript.

### PR DESCRIPTION
**Pull request by nurseiit**
* **Please check if the PR fulfills these requirements**
- [x] You have read the contribution guidelines [guidelines](https://github.com/JordanKnott/taskcafe/blob/master/CONTRIBUTING.md)
- [x] The commit message follows our [guidelines](https://github.com/JordanKnott/taskcafe/blob/master/CONTRIBUTING.md#git-commit-message-style)
- [ ] Docs have been added / updated (for bug fixes / features)

* **What kind of change does this PR introduce?** (Bug fix, feature, docs update, ...)
Refactor all legacy `moment.js` with modern `day.js`. 

* **What is the current behavior?** (You can also link to an open issue here)
Moment.js is used in frontend client for pretty much anything date related.
Fixes #74 

* **What is the new behavior (if this is a feature change)?**
N/A

* **Does this PR introduce a breaking change?** (What changes might users need to make in their application due to this PR?)
No.

* **Other information**:

### CODE REVIEW

**Review by tauqeernasir — APPROVED** [review]

**Comment by JordanKnott** [comment]
Can you squash the commits?

File: frontend/src/shared/components/Lists/index.tsx [hunk 1]
```diff
@@ -10,12 +10,15 @@ import {
...
 } from 'shared/utils/draggables';
-import moment from 'moment';
+import dayjs from 'dayjs';
+import isSameOrAfter from 'dayjs/plugin/isSameOrAfter';
 import { TaskSorting, TaskSortingType, TaskSortingDirection, sortTasks } from 'shared/utils/sorting';
 
 import { Container, BoardContainer, BoardWrapper } from './Styles';
 import shouldMetaFilter from './metaFilter';
 
+dayjs.extend(isSameOrAfter);
```
Comment (JordanKnott) [inline frontend/src/shared/components/Lists/index.tsx:20]: We should probably move this to `src/App/index.tsx`

File: frontend/src/shared/components/DueDateManager/index.tsx [hunk 2]
```diff
@@ -10,6 +11,8 @@ import NOOP from 'shared/utils/noop';
 
 import { Wrapper, ActionWrapper, RemoveDueDate, DueDateInput, DueDatePickerWrapper, ConfirmAddDueDate } from './Styles';
 
+dayjs.extend(customParseFormat);
```
Comment (JordanKnott) [inline frontend/src/shared/components/DueDateManager/index.tsx:14]: We should probably move this to `src/App/index.tsx`

File: frontend/src/shared/components/Lists/metaFilter.ts [hunk 3]
```diff
@@ -1,5 +1,10 @@
 import { TaskMetaFilters, DueDateFilterType } from 'shared/components/Lists';
-import moment from 'moment';
+import dayjs from 'dayjs';
+import weekday from 'dayjs/plugin/weekday';
+import isBetween from 'dayjs/plugin/isBetween';
+
+dayjs.extend(weekday);
+dayjs.extend(isBetween);
```
Comment (JordanKnott) [inline frontend/src/shared/components/Lists/metaFilter.ts:7]: We should probably move this to `src/App/index.tsx`

**Review by JordanKnott — CHANGES_REQUESTED** [review]
Except for squashing commits & moving some of the `extends` into `src/App/index`, everything looks good!

**Comment by JordanKnott** [comment]
Since this PR has been stale for a while, I rebased off of master & squashed the commits. I also moved the plugins to `src/index.tsx`. This PR has now bee merged.

**Pull request merged**